ops/ops/interface_kube_control/model.py

```diff
@@ -139,3 +149,21 @@ def get_ca_certificate(self, model: ops.Model) -> Optional[bytes]:
             return secret.get_content(refresh=True)["ca-certificate"].encode()
         except ops.SecretNotFoundError:
             return None
+
+
+class AuthCredentials(BaseModel):
```

Comment:
Are there any tests around getting and setting these AuthCredentials?

Reply:
I think there is. The only thing that we use from that `AuthCredentials` is the `client_token` field, and that field is used to populate the `token` in the created kubeconfig. We have a test covering it here: https://github.com/charmed-kubernetes/interface-kube-control/blob/main/ops/tests/unit/test_ops_requires.py#L157